src/main/java/de/zalando/aruha/nakadi/domain/BatchItem.java

```diff
@@ -13,6 +12,11 @@
     public BatchItem(final JSONObject event) {
         this.response = new BatchItemResponse();
         this.event = event;
+
+        Optional.
+                of(event.optJSONObject("metadata"))
+                .map(e -> e.optString("eid"))
```

Comment:
`e.optString("eid")` will return empty string is "eid" field is not found. As result the next statement (ifPresent(....)) will always be executed.
I think if you change it like `.map(e -> e.optString("eid", null))` it will work correct